version: '3.8'
services:
  db:
    container_name: postgres
    image: postgres:15
    environment:
      POSTGRES_DB: shedlock_db
      POSTGRES_USER: shedlock_user
      POSTGRES_PASSWORD: shedlock_pass
    ports:
      - "5432:5432"
    volumes:
      - postgres-data:/var/lib/postgresql/data
      - ./db-init.sql:/docker-entrypoint-initdb.d/db-init.sql
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U shedlock_user -d shedlock_db"]
      interval: 5s
      timeout: 5s
      retries: 10

  app1:
    build: .
    depends_on:
      db:
        condition: service_healthy
    environment:
      SPRING_DATASOURCE_URL: jdbc:postgresql://db:5432/shedlock_db
      SPRING_DATASOURCE_USERNAME: shedlock_user
      SPRING_DATASOURCE_PASSWORD: shedlock_pass
    ports:
      - "8081:8080"

  app2:
    build: .
    depends_on:
      db:
        condition: service_healthy
    environment:
      SPRING_DATASOURCE_URL: jdbc:postgresql://db:5432/shedlock_db
      SPRING_DATASOURCE_USERNAME: shedlock_user
      SPRING_DATASOURCE_PASSWORD: shedlock_pass
    ports:
      - "8082:8080"

volumes:
  postgres-data: